.github/workflows/config.yml

```
@@ -0,0 +1,51 @@
+name: config
+
+on:
+  push:
+    branches:
+      - master
+      - test
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - id: cache
+        uses: actions/cache@v2
+        with:
+          path: ./node_modules
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+      - run: npm install --ignore-engines
+      - run: npm run build
+      # - uses: actions/upload-artifact@v2
+      #   with:
+      #     name: node_modules
+      #     path: ./node_modules
+      - uses: actions/cache@v2
+        if: success()
+        with:
+          path: ./node_modules
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - id: cache
+        uses: actions/cache@v2
+        with:
+          path: ./node_modules
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+      - run: npm run lint
+
+  release:
+    runs-on: ubuntu-latest
```

```suggestion
  release:
    runs-on: ubuntu-latest
    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```

this makes sure we only release on pushes to the master branch